src/nupic/support/decorators.py

```diff
@@ -195,8 +194,8 @@ def myFunctionFoo():
   assert isinstance(retryExceptions, tuple), (
     "retryExceptions must be tuple, but got %r") % (type(retryExceptions),)
   
-  if getLoggerCallback is None:
-    getLoggerCallback = logging.getLogger
+  if logger is None:
+    logger = logging.getLogger(__name__)
```

Comment:
again, it may make sense to create this once at the module level rather than re-creating it each function call.